src/mapper/pkg/intentsstore/holder.go

```diff
@@ -129,27 +155,32 @@ func (i *IntentsHolder) AddIntent(newTimestamp time.Time, intent model.Intent) {
 	i.addIntentToStore(i.sinceLastGetStore, newTimestamp, intent)
 }
 
-func (i *IntentsHolder) GetIntents(namespaces []string, includeLabels []string, includeAllLabels bool) []TimestampedIntent {
+func (i *IntentsHolder) GetIntents(namespaces []string, includeLabels []string, excludeServiceWithLabels []string, includeAllLabels bool) []TimestampedIntent {
 	i.lock.Lock()
 	defer i.lock.Unlock()
 
-	return i.getIntentsFromStore(i.accumulatingStore, namespaces, includeLabels, includeAllLabels)
+	return i.getIntentsFromStore(i.accumulatingStore, namespaces, includeLabels, excludeServiceWithLabels, includeAllLabels)
 }
 
 func (i *IntentsHolder) GetNewIntentsSinceLastGet() []TimestampedIntent {
 	i.lock.Lock()
 	defer i.lock.Unlock()
 
-	intents := i.getIntentsFromStore(i.sinceLastGetStore, nil, nil, false)
+	intents := i.getIntentsFromStore(i.sinceLastGetStore, nil, nil, nil, false)
 	i.sinceLastGetStore = make(IntentsStore)
 	return intents
 }
 
-func (i *IntentsHolder) getIntentsFromStore(store IntentsStore, namespaces []string, includeLabels []string, includeAllLabels bool) []TimestampedIntent {
+func (i *IntentsHolder) getIntentsFromStore(store IntentsStore, namespaces []string, includeLabels []string, excludeServiceWithLabels []string, includeAllLabels bool) []TimestampedIntent {
 	namespacesSet := goset.FromSlice(namespaces)
 	includeLabelsSet := goset.FromSlice(includeLabels)
 	result := make([]TimestampedIntent, 0)
+
 	for pair, intent := range store {
+		if len(excludeServiceWithLabels) != 0 && intent.containsExcludedLabels(excludeServiceWithLabels) {
```

Comment:
Create the excludeServiceWithLabels map here, instead of recreating it for each intent